# Analizando aplicativos da App Store e Google Play

O objetivo deste projeto é analisar e identificar perfis de aplicativos móveis da App Store e da Google Play. Vamos trabalhar com informações que possam auxiliar na tomada de decisões sobre desenvolvimento de aplicativos.

Imagine um cenário em que produzimos aplicativos gratuitos, cuja principal fonte de renda são anúncios dentro deles. Portanto, nossa receita depende diretamente do número de usuários que usam nossos apps. Vamos examinar e sugerir quais tipos de aplicativos têm mais chances de atrair usuários.

## Abrindo e explorando os dados

Dado que a quantidade de aplicativos iOS e Android ultrapassam a casa dos milhões, estaríamos falando de um processo extremamente trabalhoso e custoso para fazer uma análise em cima deles. Sendo assim, faremos nosso trabalho com base em uma amostra desses números. Amostras essas disponíveis no [Kaggle](https://www.kaggle.com). 

A amostra sobre a Google Play contém dados de mais de 10.000 aplicativos e pode ser baixada [aqui](https://www.kaggle.com/datasets/lava18/google-play-store-apps/data).

A amostra sobre a App Store contém dados de mais de 7.000 aplicativos e pode ser baixada [aqui](https://www.kaggle.com/datasets/ramamet4/app-store-apple-data-set-10k-apps).

Vamos começar abrindo os arquivos para depois explorar os dados:

In [1]:
from csv import reader

### Google Play ###
opened_file = open('googleplaystore.csv')
read_file = reader(opened_file)
android = list(read_file)
android_header = android[0]
android = android[1:]

### App Store ###
opened_file = open('AppleStore.csv')
read_file = reader(opened_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]

Vamos agora definir uma função que irá nos ajudar a explorar e visualizar os dados:

In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n')

    if rows_and_columns:
        print('Número de linhas:', len(dataset))
        print('Número de colunas:', len(dataset[0]))

Vamos agora imprimir os headers e usar nossa função para verificarmos como estão os dados que foram importados. Note que será usado o parâmetro `True` para que possamos também conferir o número de linhas e colunas de cada amostra:

#### Android

In [3]:
print(android_header)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


In [4]:
explore_data(android, 0, 5, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


Número de linhas: 10841
Número de colunas: 13


Podemos aqui verificar algumas coisas:
- Temos um total de 10841 linhas e 13 colunas.
- As colunas que podem ser úteis ao nosso propósito são `App`, `Category`, `Genre`, `Reviews`, `Installs`, `Type` e `Price`

#### iOS

In [5]:
print(ios_header)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


In [6]:
explore_data(ios, 0, 5, True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


Número de linhas: 7197
Número de colunas: 16


Podemos aqui verificar algumas coisas:
- Temos um total de 7197 linhas e 16 colunas
- Os nomes das colunas não são auto-explicativos, nos levando a verificar sua documentação na [fonte](https://www.kaggle.com/datasets/ramamet4/app-store-apple-data-set-10k-apps)
- Após verificarmos a fonte, as colunas que devem ser úteis ao nosso propósito são `track_name`, `currency`, `price`, `rating_count_tot`, `rating_count_ver` e `prime_genre`

## Deletando dados incorretos

Nas nossas fontes temos uma área dedicada à discussão sobre elas, onde há usuários relatando suas experiências com os csvs. Lá são relatadas linhas que contêm erros e podem não ser úteis para nós. Vamos verificar:

#### Android

Um usuário relatou erro na linha 10472, sem o header, no qual a falta de informação de categoria acarretou em um deslocamento nas subsequentes. Vamos verificar:

In [7]:
print(android_header)
print('\n')
print(android[10472]) # linha com erro relatado
print('\n')
print(android[0]) # linha que sabemos que está correta, para efeito comparativo

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


Notamos que por pular a coluna `Category`, acabamos tendo um aplicativo com avaliação 19, que está acima da nota máxima da loja. A nota máxima é 5. Sendo assim, vamos deletar esta linha de nossos dados: 

In [8]:
print(len(android))
del android[10472] # não execute esta célula mais de uma vez.
print(len(android))

10841
10840


#### iOS

Não foi encontrado nenhum erro nos dados. 

## Removendo entradas duplicadas

#### Android

Nos dados da Google Play nós temos alguns lançamentos duplicados para aplicativos. Vamos encontrar quais são e definir uma maneira de atacar o problema:

In [9]:
apps_duplicados = []
apps_unicos = []

for app in android:
    nome = app[0]
    if nome in apps_unicos:
        apps_duplicados.append(nome)
    else:
        apps_unicos.append(nome)
        
print("Número de aplicativos duplicados:", len(apps_duplicados))
print("Exemplos de aplicativos duplicados:", apps_duplicados[:15])

Número de aplicativos duplicados: 1181
Exemplos de aplicativos duplicados: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


Notamos aqui que temos 1181 lançamentos duplicados em nossos dados, o que é indesejado para nosso estudo. Isto nos levanta a seguinte questão: como deletamos as linhas duplicadas? Mantemos a primeira e deletamos os próximos dados? Deletamos aleatoriamente? Vamos oberservar a busca feita pelo primeiro aplicativo da lista de lançamentos duplicados:

In [10]:
for app in android:
    name = app[0]
    if name == 'Quick PDF Scanner + OCR FREE':
        print(app)
        print("\n")

['Quick PDF Scanner + OCR FREE', 'BUSINESS', '4.2', '80805', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'February 26, 2018', 'Varies with device', '4.0.3 and up']


['Quick PDF Scanner + OCR FREE', 'BUSINESS', '4.2', '80805', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'February 26, 2018', 'Varies with device', '4.0.3 and up']


['Quick PDF Scanner + OCR FREE', 'BUSINESS', '4.2', '80804', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'February 26, 2018', 'Varies with device', '4.0.3 and up']




Podemos ver que nos dados ainda não alterados ele aparece três vezes, sem muitas variações em cada lançamento. No entanto, observando a quarta coluna podemos perceber que há uma diferença no número de `Reviews`, sendo que há um lançamento com menos avaliações. Então, podemos definir uma forma de resolver o problema: vamos manter os lançamentos com o maior número de avaliações (isso indica que esse lançamento tende a ser mais recente) e descartar o restante.

Para tal iremos criar um dicionário cuja key é um nome único de aplicativo e seu valor será o maior número de reviews que ele recebeu:

In [11]:
reviews_max = {}

for app in android:
    nome = app[0]
    n_reviews = float(app[3])

    if nome in reviews_max and n_reviews > reviews_max[nome]:
        reviews_max[nome] = n_reviews
    elif nome not in reviews_max:
        reviews_max[nome] = n_reviews

Para conferir se nosso código funcionou corretamente, podemos verificar o tamanho do dicionário que acabamos de criar. Nós tínhamos um total de 10.840 entradas sendo que 1.181 eram repetidas. Logo, nossa expectativa é que tenhamos 10.840 - 1.181 = 9659 aplicativos distintos. Vejamos:

In [12]:
print("Número esperado de aplicativos únicos:", len(android) - len(apps_duplicados))
print("Quantidade de entradas únicas no dicionário:", len(reviews_max))

Número esperado de aplicativos únicos: 9659
Quantidade de entradas únicas no dicionário: 9659


Para obtermos dados limpos de lançamentos duplicados, criaremos duas listas vazias. `android_limpo` que será uma lista de listas que receberá todas as entradas únicas e com o maior valor em `Reviews` e também `ja_adicionado` que será uma lista para verificarmos quais aplicativos já incluímos. Iremos percorrer todas as linhas dos dados `android` e checaremos tanto se aquele número na coluna `Reviews` é a maior entrada (temos isso guardado em nosso dicionário `reviews_max`) quanto também se já incluímos ou não esse app, usando nossa lista de controle. Essa última condição é importante usar pois caso tenhamos um app com lançamento duplicado mas com um número igual em `Review`, nós adicionaremos esse lançamento mais de uma vez. Para isso, lembre do exemplo que demos do aplicativo ‘Quick PDF Scanner + OCR FREE’, onde tínhamos um lançamento menor e dois iguais.

In [13]:
android_limpo = []
ja_adicionado = []

for app in android:
    nome = app[0]
    n_reviews = float(app[3])
    
    if reviews_max[nome] == n_reviews and nome not in ja_adicionado:
        android_limpo.append(app)
        ja_adicionado.append(nome)

Vamos verificar agora se nossos dados estão como esperado:

In [14]:
explore_data(android_limpo, 0, 5, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


Número de linhas: 9659
Número de colunas: 13


#### iOS

Vamos repetir o mesmo processo para verificarmos se temos lançamentos duplicados na App Store:

In [15]:
apps_duplicados = []
apps_unicos = []

for app in ios:
    nome = app[0]
    if nome in apps_unicos:
        apps_duplicados.append(nome)
    else:
        apps_unicos.append(nome)
        
print("Número de aplicativos duplicados:", len(apps_duplicados))
print("Exemplos de aplicativos duplicados:", apps_duplicados[:10])

Número de aplicativos duplicados: 0
Exemplos de aplicativos duplicados: []


Como não temos lançamentos duplicados, nada a fazer por aqui.

## Removendo aplicativos que não estão em inglês

Vamos nos concentrar nos aplicativos que foram desenvolvidos para o público que fala inglês (considerando que a maioria dos apps tem seu nome nesse idioma e que as outras colunas também estão nessa língua). Alguns aplicativos têm nomes em outros idiomas que não podemos processar e entender corretamente. Vejamos alguns exemplos:

In [40]:
print(ios[813][1])
print(ios[6731][1])
print(android_limpo[4412][0])
print(android_limpo[7940][0])

爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜
中国語 AQリスニング
لعبة تقدر تربح DZ


Para filtrar esses apps, vamos usar o sistema [ASCII](https://en.wikipedia.org/wiki/ASCII). Os caracteres usados nesse idioma estão entre 0 e 127, de acordo com a tabela. No entanto, podemos ter o problema de que alguns aplicativos têm emojis ou outros símbolos em seus nomes (que escapam do intervalo de 0 a 127 da tabela ASCII), gerando perdas que podem ser valiosas. Vejamos um exemplo que pudemos detectar, na impressão anterior, logo que fizemos nossa lista limpa dos apps androids:

In [17]:
print("Nome do app:", android_limpo[1][0])
print("ASCII – :", ord('–'))

Nome do app: U Launcher Lite – FREE Live Cool Themes, Hide Apps
ASCII – : 8211


Um jeito de evitar apagar apps por esses caracteres será o de só remover dos dados aqueles aplicativos que têm mais de 3 caracteres fora do intervalo 0 e 127. Esse método irá filtrar a maior parte dos aplicativos indesejados e será bem eficaz.

Começaremos com uma função para detectar se está em inglês ou não e vamos testar no app que já mencionamos antes:

In [18]:
def esta_em_ingles(string):
    fora_ascii = 0
    
    for caractere in string:
        if ord(caractere) > 127:
            fora_ascii += 1
    
    if fora_ascii > 3:
        return False
    else:
        return True
    
print(esta_em_ingles(android_limpo[1][0]))

True


Com nossa função pronta, vamos agora criar duas novas listas de listas vazias e preenchê-las apenas com os aplicativos desejados:

In [19]:
android_ingles = []
ios_ingles = []

for app in android_limpo:
    nome = app[0]
    if esta_em_ingles(nome):
        android_ingles.append(app)
        
for app in ios:
    nome = app[1]
    if esta_em_ingles(nome):
        ios_ingles.append(app)

Vamos agora imprimir parte desses dados e verificar:

In [20]:
explore_data(android_ingles, 0, 3, True)
print("\n")
explore_data(ios_ingles, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Número de linhas: 9614
Número de colunas: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+'

Note que alguns apps foram filtrados:

In [21]:
print("Apps Androis filtrados:", len(android_limpo) - len(android_ingles))
print("Apps iOS filtrados:", len(ios) - len(ios_ingles))

Apps Androis filtrados: 45
Apps iOS filtrados: 1014


## Aplicativos gratuitos

Como mencionamos anteriormente, queremos analisar apenas os aplicativos gratuitos. Para tal, vamos criar duas listas vazias e preencher elas com as informações dos aplicativos grátis:

In [22]:
android_final = []
ios_final = []

for app in android_ingles:
    preco = app[7]
    if preco == '0':
        android_final.append(app)

for app in ios_ingles:
    preco = app[4]
    if preco == '0.0':
        ios_final.append(app)

Vamos imprimir parte dos dados e verificar:

In [23]:
explore_data(android_final, 0, 3, True)
print("\n")
explore_data(ios_final, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Número de linhas: 8864
Número de colunas: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+'

Consideraremos algumas hipóteses:

- Primeiro, desenvolveremos nossos apps para uma das plataformas, por exemplo, Google Play.
- Depois de analisar sua aceitação, iremos melhorar mais o app e também disponibilizá-lo para a outra plataforma.

## Aplicativos mais comuns por gênero

Como queremos aplicativos que tenham sucesso nas duas lojas, vamos procurar perfis que tendem a ser bem recebidos na Google Play e na App Store de forma gratuita.

Para tal vamos começar nossa análise pensando nos gêneros mais comuns em cada mercado. Vamos fazer tabelas de frequência olhando a coluna `prime_genre` nos dados da App Store e as colunas e `Category`e `Genre` nos da Google Play.

Vamos trabalhar com duas funções:
- Uma para gerar a tabela de frequência mostrando porcentagem
- Outra para ordenar em ordem decrescente e mostrar as tabelas de porcentagem

In [24]:
def tabela_frequencia(data, indice):
    tabela = {}
    total = 0
    
    for row in data:
        total += 1
        valor = row[indice]
        if valor in tabela:
            tabela[valor] += 1
        else:
            tabela[valor] = 1
    
    tabela_percentual = {}
    for key in tabela:
        percentual = (tabela[key] / total) * 100
        tabela_percentual[key] = percentual 
    
    return tabela_percentual


def mostra_tabela(data, indice):
    tabela = tabela_frequencia(data, indice)
    tabela_mostra = []
    for key in tabela:
        key_val_como_tupla = (tabela[key], key)
        tabela_mostra.append(key_val_como_tupla)
        
    tabela_ordenada = sorted(tabela_mostra, reverse=True)
    for entrada in tabela_ordenada:
        print(entrada[1], ':', round(entrada[0], 2), '%')

Feito isso, vamos analisar os resultados na AppStore:

In [25]:
mostra_tabela(ios_final, -5)

Games : 58.16 %
Entertainment : 7.88 %
Photo & Video : 4.97 %
Education : 3.66 %
Social Networking : 3.29 %
Shopping : 2.61 %
Utilities : 2.51 %
Sports : 2.14 %
Music : 2.05 %
Health & Fitness : 2.02 %
Productivity : 1.74 %
Lifestyle : 1.58 %
News : 1.33 %
Travel : 1.24 %
Finance : 1.12 %
Weather : 0.87 %
Food & Drink : 0.81 %
Reference : 0.56 %
Business : 0.53 %
Book : 0.43 %
Navigation : 0.19 %
Medical : 0.19 %
Catalogs : 0.12 %


Podemos observar que a categoria de jogos representa 58.16% dos aplicativos gratuitos, seguida pelas categorias de entretenimento (7.88%), foto e vídeo (4.97%), educação (3.66%) e redes sociais (3.29%).

Uma conclusão possível é que a maior parte dos aplicativos são feitos com foco na diversão do usuário. Jogos, entretenimento, foto e vídeo e redes sociais representam 74.3% do total dos aplicativos em nossa amostra. Uma parcela considerável!

Agora façamos o mesmo para os aplicativos gratuitos da Play Store primeiro na coluna `Category`:

In [26]:
mostra_tabela(android_final, 1)

FAMILY : 18.91 %
GAME : 9.72 %
TOOLS : 8.46 %
BUSINESS : 4.59 %
LIFESTYLE : 3.9 %
PRODUCTIVITY : 3.89 %
FINANCE : 3.7 %
MEDICAL : 3.53 %
SPORTS : 3.4 %
PERSONALIZATION : 3.32 %
COMMUNICATION : 3.24 %
HEALTH_AND_FITNESS : 3.08 %
PHOTOGRAPHY : 2.94 %
NEWS_AND_MAGAZINES : 2.8 %
SOCIAL : 2.66 %
TRAVEL_AND_LOCAL : 2.34 %
SHOPPING : 2.25 %
BOOKS_AND_REFERENCE : 2.14 %
DATING : 1.86 %
VIDEO_PLAYERS : 1.79 %
MAPS_AND_NAVIGATION : 1.4 %
FOOD_AND_DRINK : 1.24 %
EDUCATION : 1.16 %
ENTERTAINMENT : 0.96 %
LIBRARIES_AND_DEMO : 0.94 %
AUTO_AND_VEHICLES : 0.93 %
HOUSE_AND_HOME : 0.82 %
WEATHER : 0.8 %
EVENTS : 0.71 %
PARENTING : 0.65 %
ART_AND_DESIGN : 0.64 %
COMICS : 0.62 %
BEAUTY : 0.6 %


Podemos perceber aqui um comportamento um pouco diferente quando se trata da loja para Android. Vemos uma distribuição menor focada na diversão do usuário e mais em aplicativos que parecem ser focados em serem úteis (família, ferramentas, negócios, produtividade e estilo de vida). No entanto, vale a pena olhar mais de perto a categoria família, que foi a com maior número de aplicativos:

[<img src="family.jpg">](https://play.google.com/store/apps/category/FAMILY?hl=en)

Observando a categoria na Play Store (agora renomeada para Kids), vemos que ela se trata, em sua maior parte, de jogos infantis.

Mesmo assim, podemos notar que as categorias ferramentas, negócios, estilo de vida, produtividade, finanças, saúde e outras que reúnem praticidades e utilidades continuam representando uma fatia muito maior que a loja da Apple.

Nos nossos dados da Google Play nós temos também a coluna `Genres`. Vamos olhar para confirmar:

In [27]:
mostra_tabela(android_final, -4)

Tools : 8.45 %
Entertainment : 6.07 %
Education : 5.35 %
Business : 4.59 %
Productivity : 3.89 %
Lifestyle : 3.89 %
Finance : 3.7 %
Medical : 3.53 %
Sports : 3.46 %
Personalization : 3.32 %
Communication : 3.24 %
Action : 3.1 %
Health & Fitness : 3.08 %
Photography : 2.94 %
News & Magazines : 2.8 %
Social : 2.66 %
Travel & Local : 2.32 %
Shopping : 2.25 %
Books & Reference : 2.14 %
Simulation : 2.04 %
Dating : 1.86 %
Arcade : 1.85 %
Video Players & Editors : 1.77 %
Casual : 1.76 %
Maps & Navigation : 1.4 %
Food & Drink : 1.24 %
Puzzle : 1.13 %
Racing : 0.99 %
Role Playing : 0.94 %
Libraries & Demo : 0.94 %
Auto & Vehicles : 0.93 %
Strategy : 0.91 %
House & Home : 0.82 %
Weather : 0.8 %
Events : 0.71 %
Adventure : 0.68 %
Comics : 0.61 %
Beauty : 0.6 %
Art & Design : 0.6 %
Parenting : 0.5 %
Card : 0.45 %
Casino : 0.43 %
Trivia : 0.42 %
Educational;Education : 0.39 %
Board : 0.38 %
Educational : 0.37 %
Education;Education : 0.34 %
Word : 0.26 %
Casual;Pretend Play : 0.24 %
Music : 0.2 %
R

Podemos notar duas coisas da impressão anterior:

- Temos aqui uma tabela com muito mais categorias
- Nossas suspeitas de que os aplicativos gratuitos de utilidade e praticidade são mais frequentes na loja se confirmam. Podemos observar uma grande quantidade em ferramentas, educação, negócios, produtividade, estilo de vida e etc.

Como consequência da grande quantidade de categorias, vamos focar na opção anterior que nos trará uma maior variedade de apps dentro de cada uma. Estamos querendo olhar o panorama geral.

## Aplicativos mais populares por gênero na App Store

Para sabermos quais gêneros são mais populares, nós podemos calcular a média de instalação para cada gênero de aplicativo. Para isso, nós iremos:

- Separar os aplicativos de cada gênero
- Somar o número de instalações de aplicativos para cada gênero
- Dividir essa soma pelo número de aplicativos naquele gênero

Quando olhamos o cabeçalho de nossos dados na App Store, encontramos um problema: não temos o número de instalações disponível. Para os dados da Google Play, temos a coluna `Installs`, mas aqui não. Para contornar essa situação, vamos usar o que temos disponível: a coluna `rating_count_tot`. Essa coluna se refere ao número total de avaliações, não é uma medida perfeita, mas servirá para o nosso propósito de calcular o número de avaliações médio por aplicativo em cada gênero.

In [28]:
generos_ios = tabela_frequencia(ios_final, -5)
gen_avg_ios = []

for genero in generos_ios:
    total = 0
    len_genero = 0
    for app in ios_final:
        genero_app = app[-5]
        if genero_app == genero:            
            n_ratings = float(app[5])
            total += n_ratings
            len_genero += 1
    avg_n_ratings = total / len_genero
    print(genero, ':', round(avg_n_ratings, 2))

Social Networking : 71548.35
Photo & Video : 28441.54
Games : 22788.67
Music : 57326.53
Reference : 74942.11
Health & Fitness : 23298.02
Weather : 52279.89
Utilities : 18684.46
Travel : 28243.8
Shopping : 26919.69
News : 21248.02
Navigation : 86090.33
Lifestyle : 16485.76
Entertainment : 14029.83
Food & Drink : 33333.92
Sports : 23008.9
Book : 39758.5
Finance : 31467.94
Education : 7003.98
Productivity : 21028.41
Business : 7491.12
Catalogs : 4004.0
Medical : 612.0


Na média, os aplicativos de navegação têm o maior número de avaliações, seguidos pelos gêneros de referências, redes sociais e músicas. Vamos conferir:

In [29]:
for app in ios_final:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


Se continuarmos essa avaliação para redes sociais e músicas, veremos o mesmo comportamento se repetindo. Gigantes como Facebook, Pinterest, Pandora, Spotify e Shazam acabam por elevar bastante a média. Nosso objetivo é buscar por gêneros populares, mas esses poucos casos isolados acabam deixando a categoria mais populares do que realmente são e também tornam a competição extremamente difícil, seja em concorrência ou seja em custos de criação e manutenção.

Clima, apesar de popular, pode não ser uma escolha muito boa dadas as nossas hipóteses. Esses aplicativos não costumam manter o usuário por muito tempo, diminuindo nossa quantidade de anúncios exibidos e também ter dados bons e confiáveis sobre a previsão pode nos levar a APIs pagas. O problema com isso é que nós queríamos pensar primeiro em testar a rentabilidade das aplicações para depois trabalharmos mais no seu desenvolvimento, gerando um custo que não necessariamente queremos ter logo no início.

Jogos são muito populares em ambas categorias e se encaixam no propósito de manter jogadores por muito tempo e aumentando nossa quantidade de anúncios exibidos. É uma recomendação válida e um tanto quanto esperada. Ao olharmos a distribuição de avaliações por jogos, notamos que eles não seguem o mesmo padrão das categorias que falamos anteriormente, isto é, não temos uns poucos aplicativos tornando a categoria mais popular do que realmente é. Vejamos apenas a fatia acima de 100.000 avaliações:

In [52]:
for app in ios_final:
    if app[-5] == 'Games' and float(app[5]) > 100000:
        print(app[1], ':', app[5])

Clash of Clans : 2130805
Temple Run : 1724546
Candy Crush Saga : 961794
Angry Birds : 824451
Subway Surfers : 706110
Solitaire : 679055
CSR Racing : 677247
Crossy Road - Endless Arcade Hopper : 669079
Injustice: Gods Among Us : 612532
Hay Day : 567344
PAC-MAN : 508808
DragonVale : 503230
Head Soccer : 481564
Despicable Me: Minion Rush : 464312
The Sims™ FreePlay : 446880
Sonic Dash : 418033
8 Ball Pool™ : 416736
Tiny Tower - Free City Building : 414803
Jetpack Joyride : 405647
Bike Race - Top Motorcycle Racing Games : 405007
Kim Kardashian: Hollywood : 397730
Trivia Crack : 393469
WordBrain : 391401
Sniper 3D Assassin: Shoot to Kill Gun Game : 386521
Flow Free : 373857
Geometry Dash Lite : 370370
▻Sudoku : 359832
Fruit Ninja® : 327025
Pixel Gun 3D : 301182
Temple Run 2 : 295211
My Horse : 293857
Word Cookies! : 287095
Dragon City Mobile : 277268
The Simpsons™: Tapped Out : 274501
Plants vs. Zombies™ 2 : 267394
Clash Royale : 266921
Pokémon GO : 257627
CSR Racing 2 : 257100
Star Wars™: 

Agora vamos olhar para as próximas: Comida e bebida, referências e livros.

In [30]:
for app in ios_final:
    if app[-5] == 'Food & Drink':
        print(app[1], ':', app[5])

Starbucks : 303856
Domino's Pizza USA : 258624
OpenTable - Restaurant Reservations : 113936
Allrecipes Dinner Spinner : 109349
DoorDash - Food Delivery : 25947
UberEATS: Uber for Food Delivery : 17865
Postmates - Food Delivery, Faster : 9519
Dunkin' Donuts - Get Offers, Coupons & Rewards : 9068
Chick-fil-A : 5665
McDonald's : 4050
Deliveroo: Restaurant Delivery - Order Food Nearby : 1702
SONIC Drive-In : 1645
Nowait Guest : 1625
7-Eleven, Inc. : 1356
Outback : 805
Bon Appetit : 750
Starbucks Keyboard : 457
Whataburger : 197
Delish Eatmoji Keyboard : 154
Lieferheld - Delicious food delivery service : 29
Lieferando.de : 29
McDo France : 22
Chefkoch - Rezepte, Kochen, Backen & Kochbuch : 20
Youmiam : 9
Marmiton Twist : 2
Open Food Facts : 1


In [31]:
for app in ios_final:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


In [32]:
for app in ios_final:
    if app[-5] == 'Book':
        print(app[1], ':', app[5])

Kindle – Read eBooks, Magazines & Textbooks : 252076
Audible – audio books, original series & podcasts : 105274
Color Therapy Adult Coloring Book for Adults : 84062
OverDrive – Library eBooks and Audiobooks : 65450
HOOKED - Chat Stories : 47829
BookShout: Read eBooks & Track Your Reading Goals : 879
Dr. Seuss Treasury — 50 best kids books : 451
Green Riding Hood : 392
Weirdwood Manor : 197
MangaZERO - comic reader : 9
ikouhoushi : 0
MangaTiara - love comic reader : 0
謎解き : 0
謎解き2016 : 0


Sobre a categoria comida, podemos ver que os aplicativos populares na verdade estão focados ou em restaurantes ou entrega de comida, o que foge da nossa ideia inicial que é nos limitarmos a somente aplicativos, sem outros negócios por trás.

Quando nos deparamos com os aplicativos de referência ou livros, podemos notar o padrão de que os mais populares estão ligados a reprodução de livros/audiolivros e também livros ou dicionários propriamente ditos. Aqui temos algo interessante, nos dando oportunidade de pensar tanto em fazermos uma mídia digital de alguma obra popular ou então trabalharmos em bibliotecas ou reprodutores. Até mesmo guias ou mapas para games mostram alguma evidência aqui. Além disso, vamos lembrar que nossa renda virá da exibição de anúncios e esse estilo de aplicativos costumam manter o usuário por bastante tempo enquanto consomem seu conteúdo, ou seja, mais anúncios serão exibidos.

## Aplicativos mais populares por gênero na Google Play

Nestes dados nós temos os números referentes às instalações porém seu formato dificulta um pouco a análise. Vejamos:

In [33]:
mostra_tabela(android_final, 5)

1,000,000+ : 15.73 %
100,000+ : 11.55 %
10,000,000+ : 10.55 %
10,000+ : 10.2 %
1,000+ : 8.39 %
100+ : 6.92 %
5,000,000+ : 6.83 %
500,000+ : 5.56 %
50,000+ : 4.77 %
5,000+ : 4.51 %
10+ : 3.54 %
500+ : 3.25 %
50,000,000+ : 2.3 %
100,000,000+ : 2.13 %
50+ : 1.92 %
5+ : 0.79 %
1+ : 0.51 %
500,000,000+ : 0.27 %
1,000,000,000+ : 0.23 %
0+ : 0.05 %
0 : 0.01 %


Observe que a distribuição está por intervalos e não por número exato. Ou seja, se um aplicativo tem 100.000+ instalações, não sabemos afirmar exatamente quantas são, apenas que está entre 100.000 e 1.000.000. Para continuarmos, vamos trabalhar o número inicial do intervalo, isto é, vamos considerar que 100.000+ = 100.000. Vamos fazer:

In [34]:
categorias_android = tabela_frequencia(android_final, 1)

for categoria in categorias_android:
    total = 0
    len_categoria = 0
    for app in android_final:
        categoria_app = app[1]
        if categoria_app == categoria:            
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_categoria += 1
    avg_n_installs = total / len_categoria
    print(categoria, ':', round(avg_n_installs, 2))

ART_AND_DESIGN : 1986335.09
AUTO_AND_VEHICLES : 647317.82
BEAUTY : 513151.89
BOOKS_AND_REFERENCE : 8767811.89
BUSINESS : 1712290.15
COMICS : 817657.27
COMMUNICATION : 38456119.17
DATING : 854028.83
EDUCATION : 1833495.15
ENTERTAINMENT : 11640705.88
EVENTS : 253542.22
FINANCE : 1387692.48
FOOD_AND_DRINK : 1924897.74
HEALTH_AND_FITNESS : 4188821.99
HOUSE_AND_HOME : 1331540.56
LIBRARIES_AND_DEMO : 638503.73
LIFESTYLE : 1437816.27
GAME : 15588015.6
FAMILY : 3695641.82
MEDICAL : 120550.62
SOCIAL : 23253652.13
SHOPPING : 7036877.31
PHOTOGRAPHY : 17840110.4
SPORTS : 3638640.14
TRAVEL_AND_LOCAL : 13984077.71
TOOLS : 10801391.3
PERSONALIZATION : 5201482.61
PRODUCTIVITY : 16787331.34
PARENTING : 542603.62
WEATHER : 5074486.2
VIDEO_PLAYERS : 24727872.45
NEWS_AND_MAGAZINES : 9549178.47
MAPS_AND_NAVIGATION : 4056941.77


Como vimos anteriormente na App Store, as categorias comunicação, reprodutores de vídeos, redes sociais, foto e produtividade seguem o padrão de serem muito influenciadas por alguns poucos aplicativos com muitas instalações. Vejamos um exemplo onde remover alguns poucos aplicativos acima de 100 milhões reduzem muito a média em comunicação:

In [35]:
for app in android_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

In [36]:
abaixo_100_m = []

for app in android_final:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'COMMUNICATION') and (float(n_installs) < 100000000):
        abaixo_100_m.append(float(n_installs))
        
print(round(sum(abaixo_100_m) / len(abaixo_100_m), 2))

3603485.39


Ou seja, a média cai de 38.456.119,17 para 3.603.485,39. Reduzida para menos de 10% da original. 

Vamos voltar agora nosso foco para as categorias que já havíamos recomendado na App Store, dado que estamos pensando em um aplicativo para ser desenvolvido em ambas plataformas.

Analisando os jogos, podemos ver um grande número de aplicativos nas três camadas superiores (e o mesmo se repete se filtrarmos para qualquer camada), nos mostrando uma boa distribuição e também confirmando o que já tínhamos falado anteriormente. Vamos manter essa indicação.

In [49]:
for app in android_final:
    if app[1] == 'GAME' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Sonic Dash : 100,000,000+
PAC-MAN : 100,000,000+
Roll the Ball® - slide puzzle : 100,000,000+
Piano Tiles 2™ : 100,000,000+
Pokémon GO : 100,000,000+
Extreme Car Driving Simulator : 100,000,000+
Trivia Crack : 100,000,000+
Angry Birds 2 : 100,000,000+
Candy Crush Saga : 500,000,000+
8 Ball Pool : 100,000,000+
Subway Surfers : 1,000,000,000+
Candy Crush Soda Saga : 100,000,000+
Clash Royale : 100,000,000+
Clash of Clans : 100,000,000+
Plants vs. Zombies FREE : 100,000,000+
Pou : 500,000,000+
Flow Free : 100,000,000+
My Talking Angela : 100,000,000+
slither.io : 100,000,000+
Cooking Fever : 100,000,000+
Yes day : 100,000,000+
Score! Hero : 100,000,000+
Dream League Soccer 2018 : 100,000,000+
My Talking Tom : 500,000,000+
Sniper 3D Gun Shooter: Free Shooting Games - FPS : 100,000,000+
Zombie Tsunami : 100,000,000+
Helix Jump : 100,000,000+
Crossy Road : 100,000,000+
Temple Run 2 : 500,000,000+
Talking Tom Gold Run : 100,000,000+
Agar.io : 100,000,000+
Bus Rush: Subway Edition : 100,000,00

Em livros e referências, temos:

In [37]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5]) 

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

Vamos filtrar as duas maiores faixas para termos mais clareza na visualização:

In [38]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'):
        print(app[0], ':', app[5])    

Google Play Books : 1,000,000,000+


Encontramos apenas um aplicativo, Google Play Books. Vejamos o que ocorre em uma busca em todos os aplicativos acima de 1 milhão de instalações:

In [39]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'
                                            or app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Aldiko Book Reader : 10,000,000+
Wattpad 📖 Free Books : 100,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al ka

Aqui percebemos um padrão um pouco semelhante ao que vimos na App Store: alguns livros/dicionários específicos e muitos aplicativos voltados para ler livros, audiolivros e similares. É uma recomendação adequada ao nosso objetivo e vamos mantê-la também.

## Conclusão

Neste projeto, analisamos e identificamos perfis de aplicativos móveis gratuitos da App Store e da Google Play que são populares entre os usuários. Nosso objetivo era ajudar na tomada de decisões sobre o desenvolvimento de aplicativos, considerando que nossa fonte de renda são anúncios dentro dos aplicativos.

Para isso, usamos dados de amostras de ambas as lojas e aplicamos técnicas de limpeza, filtragem e análise dos dados. Também usamos o sistema ASCII para lidar com caracteres especiais nos nomes dos aplicativos e calculamos a média de instalações e avaliações por gênero e categoria.

Nossas principais recomendações foram:

- Jogos: são muito populares em ambas as lojas e se encaixam no propósito de manter os usuários por muito tempo, aumentando a quantidade de anúncios exibidos. Podemos explorar diferentes gêneros e estilos de jogos, desde que sejam divertidos e envolventes.
- Livros e referências: são populares na Google Play e têm potencial na App Store. Podemos pensar em fazer uma mídia digital de alguma obra popular ou trabalhar em bibliotecas ou reprodutores de livros e audiolivros. Também podemos incluir recursos extras, como guias ou mapas para games, dicionários ou tradutores. Esses aplicativos costumam manter os usuários por bastante tempo enquanto consomem o conteúdo, gerando mais anúncios.